app/src/test/kotlin/com/intive/picover/images/repository/ImagesRepositoryTest.kt

```diff
@@ -0,0 +1,104 @@
+package com.intive.picover.images.repository
+
+import android.net.Uri
+import com.google.android.gms.tasks.OnFailureListener
+import com.google.android.gms.tasks.OnSuccessListener
+import com.google.android.gms.tasks.Task
+import com.google.firebase.storage.FirebaseStorage
+import com.google.firebase.storage.ListResult
+import com.google.firebase.storage.StorageReference
+import io.kotest.assertions.throwables.shouldThrowExactly
+import io.kotest.core.spec.IsolationMode
+import io.kotest.core.spec.style.ShouldSpec
+import io.kotest.matchers.shouldBe
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.slot
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.StandardTestDispatcher
+import kotlinx.coroutines.test.runTest
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class ImagesRepositoryTest : ShouldSpec({
+	isolationMode = IsolationMode.InstancePerTest
```

Comment:
i think we over-complicated this a little :D 

I would just declare tested value once and get rid off 

isolationMode = IsolationMode.InstancePerTest

and beforeSpec and beforeEach

something like:

`val testDispatcher = StandardTestDispatcher()
		val storageReference = mockk<StorageReference>(relaxed = true)
		val firebaseStorage = mockk<FirebaseStorage> {
			every { reference.child("image") } returns storageReference
		}
		val storageItem = mockk<StorageReference>()
		val tested = ImagesRepository(firebaseStorage, testDispatcher)

		should("downloaded URIs WHEN storage reference and uri is fetched successfully") {
			runTest(testDispatcher) {
				val uriResult: Uri = mockk()
				val listResult: ListResult = mockk {
					every { items } returns listOf(storageItem)
				}
				val listResultOnSuccessListenerSlot = slot<OnSuccessListener<ListResult>>()
				val uriOnSuccessListenerSlot = slot<OnSuccessListener<Uri>>()
				val listResultTask: Task<ListResult> = mockk(relaxed = true) {
					every { addOnSuccessListener(capture(listResultOnSuccessListenerSlot)) } answers {
						listResultOnSuccessListenerSlot.captured.onSuccess(listResult)
						this@mockk
					}
				}
				every { storageReference.listAll() } returns listResultTask
				val uriTask: Task<Uri> = mockk(relaxed = true) {
					every { addOnSuccessListener(capture(uriOnSuccessListenerSlot)) } answers {
						uriOnSuccessListenerSlot.captured.onSuccess(uriResult)
						this@mockk
					}
				}
				every { storageItem.downloadUrl } returns uriTask

				tested.fetchImages() shouldBe listOf(uriResult)
			}
		}

		should("throw exception WHEN fetching storage reference fails") {
			runTest(testDispatcher) {
				val listResultOnFailureListenerSlot = slot<OnFailureListener>()
				val listResultTask: Task<ListResult> = mockk {
					every { addOnSuccessListener(any()) } returns this@mockk
					every { addOnFailureListener(capture(listResultOnFailureListenerSlot)) } answers {
						listResultOnFailureListenerSlot.captured.onFailure(Exception())
						this@mockk
					}
				}
				every { storageReference.listAll() } returns listResultTask

				shouldThrowExactly<Exception> {
					tested.fetchImages()
				}
			}
		}

		should("throw exception WHEN fetching the uri fails") {
			runTest(testDispatcher) {
				val listResult: ListResult = mockk {
					every { items } returns listOf(storageItem)
				}
				val listResultOnSuccessListenerSlot = slot<OnSuccessListener<ListResult>>()
				val uriOnFailureListenerSlot = slot<OnFailureListener>()
				val listResultTask: Task<ListResult> = mockk(relaxed = true) {
					every { addOnSuccessListener(capture(listResultOnSuccessListenerSlot)) } answers {
						listResultOnSuccessListenerSlot.captured.onSuccess(listResult)
						this@mockk
					}
				}
				every { storageReference.listAll() } returns listResultTask
				val uriTask: Task<Uri> = mockk(relaxed = true) {
					every { addOnSuccessListener(any()) } returns this@mockk
					every { addOnFailureListener(capture(uriOnFailureListenerSlot)) } answers {
						uriOnFailureListenerSlot.captured.onFailure(Exception())
						this@mockk
					}
				}
				every { storageItem.downloadUrl } returns uriTask

				shouldThrowExactly<Exception> {
					tested.fetchImages()
				}
			}
		}`